default.nix

```diff
@@ -21,17 +28,43 @@ let
   # Construct a 'base' Haskell package, disabling the test
   # and benchmark suites for all dependencies by default.
   baseHaskellPkgs =
-    pkgs.haskell.packages.${ghc}.override (hpArgs: {
+    pkgs.haskell.packages.ghc8102.override (hpArgs: {
       overrides = pkgs.lib.composeExtensions (hpArgs.overrides or (_: _: { })) (
         _hfinal: hprev: {
-          mkDerivation = args: hprev.mkDerivation (args // {
+          # This is the original mkDerivation override written by jkachmar.
+          # The args passed to this mkDerivation call always get overridden by
+          # `doCheck = false`, which causes it not to work.
+          # mkDerivation = args: hprev.mkDerivation (args // {
+          #   doCheck = false;
+          #   doBenchmark = false;
+          #   doHoogle = false;
+          #   doHaddock = false;
+          #   enableLibraryProfiling = false;
+          #   enableExecutableProfiling = false;
+          # });
+
+          # This is a mkDerivation call suggested by expipiplus1.  It lets args
+          # override the `doCheck = false` setting, which we take advantage of by
+          # setting `doCheck = true` in `shell.nix`.
+          mkDerivation = args: hprev.mkDerivation ({
             doCheck = false;
             doBenchmark = false;
             doHoogle = false;
             doHaddock = false;
             enableLibraryProfiling = false;
             enableExecutableProfiling = false;
-          });
+          } // args);
```

Comment:
This is from https://github.com/NixOS/nixpkgs/issues/104349#issuecomment-730873499